File: app/src/main/java/knf/kuma/database/BaseConverter.java
package knf.kuma.database;

import android.net.Uri;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import androidx.room.TypeConverter;
import knf.kuma.videoservers.Headers;

public class BaseConverter {
    @TypeConverter
    public int BooleanToInt(Boolean b) {
        return b ? 1 : 0;
    }

    @TypeConverter
    public Boolean intToBoolean(int i) {
        return i == 1;
    }

    @TypeConverter
    public String UriToString(Uri uri) {
        return uri.toString();
    }

    @TypeConverter
    public Uri StringToUri(String s) {
        return Uri.parse(s);
    }

    @TypeConverter
    public String HeadersToString(Headers headers) {
        return new Gson().toJson(headers, new TypeToken<Headers>() {
        }.getType());
    }

    @TypeConverter
    public Headers StringToHeader(String json) {
        return new Gson().fromJson(json, new TypeToken<Headers>() {
        }.getType());
    }
}

File: app/src/main/java/knf/kuma/database/BaseConverter.kt
package knf.kuma.database

import android.net.Uri
import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import knf.kuma.videoservers.Headers

class BaseConverter {
    @TypeConverter
    fun booleanToInt(b: Boolean): Int {
        return if (b) 1 else 0
    }

    @TypeConverter
    fun intToBoolean(i: Int): Boolean {
        return i == 1
    }

    @TypeConverter
    fun uriToString(uri: Uri): String {
        return uri.toString()
    }

    @TypeConverter
    fun stringToUri(s: String): Uri {
        return Uri.parse(s)
    }

    @TypeConverter
    fun headersToString(headers: Headers?): String {
        return Gson().toJson(headers, object : TypeToken<Headers>() {
        }.type)
    }

    @TypeConverter
    fun stringToHeader(json: String?): Headers? {
        return Gson().fromJson<Headers>(json, object : TypeToken<Headers>() {
        }.type)
    }
}
